controllers/managedocs_controller.go

```diff
@@ -714,86 +719,27 @@ func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
 			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
 		}
 
-		alertmanagerConfig := r.generateAlertmanagerConfig(pagerdutyServiceKey, dmsURL)
-		config, err := yaml.Marshal(alertmanagerConfig)
-		if err != nil {
-			return fmt.Errorf("Unable to encode alertmanager conifg: %v", err)
-		}
-		r.alertmanagerConfigSecret.Data = map[string][]byte{
-			"alertmanager.yaml": config,
+		desired := templates.AlertmanagerConfigTemplate.DeepCopy()
+		for i := range desired.Spec.Receivers {
+			receiver := &desired.Spec.Receivers[i]
+			switch receiver.Name {
+			case "pagerduty":
+				receiver.PagerDutyConfigs[0].ServiceKey.Key = "PAGERDUTY_KEY"
+				receiver.PagerDutyConfigs[0].ServiceKey.LocalObjectReference.Name = r.PagerdutySecretName
+				receiver.PagerDutyConfigs[0].Details[0].Key = "SOP"
+				receiver.PagerDutyConfigs[0].Details[0].Value = r.SOPEndpoint
+			case "DeadMansSnitch":
+				receiver.WebhookConfigs[0].URL = &dmsURL
+			}
 		}
+		r.alertmanagerConfig.Spec = desired.Spec
+		r.alertmanagerConfig.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
```

Comment:
This will override all labels on the alertmanagerconfig which means that all other labels will be lost.
Please add the label to an existing ObjectMeta.labels instead